lib/autoyast.pm

```diff
@@ -783,6 +784,48 @@ sub expand_agama_profile {
     return $profile_url;
 }
 
+=head2 install_jsonnet
+
+ install_jsonnet();
+
+ Install via zypper golang-github-google-jsonnet package in the worker.
+
+=cut
+
+sub install_jsonnet {
+    if (system("sudo", "zypper", "-n", "repos", "systemsmanagement_Agama_Devel") != 0) {
+        system("sudo", "zypper", "-n", "addrepo", "-f", "-G",
+            "https://download.opensuse.org/repositories/systemsmanagement:Agama:Devel/15.6/systemsmanagement:Agama:Devel.repo");
+    }
+
+    if (system("command", "-v", "jsonnet") != 0) {
+        system("sudo", "zypper", "-n", "install", "-f", "golang-github-google-jsonnet");
+    }
+}
+
+=head2 generate_json_profile
+
+ generate_json_profile();
+
+ Return the URL of generated JSON profile
+
+=cut
+
+sub generate_json_profile {
+    my $profile_name = "generated_profile.json";
+
+    install_jsonnet();
```

Comment:
Let's have all the initialization together, so you can move this block before jsonnet execution.